src/librustc/hir/map/definitions.rs

```diff
@@ -136,7 +136,9 @@ impl DefKey {
 
         ::std::mem::discriminant(data).hash(&mut hasher);
         if let Some(name) = data.get_opt_name() {
-            name.hash(&mut hasher);
+            // Get a stable hash by considering the symbol chars rather than
+            // the symbol index.
+            name.as_str().hash(&mut hasher);
```

Comment:
Is there no more straight-forward `.stable_hash` or similar?

Reply:
There is `Symbol::hash_stable` but it requires a `StableHashingContext` argument, even though it doesn't use that argument. This code is equivalent.